src/common.c

```diff
@@ -1418,16 +1418,81 @@ sr_create_module_imps_incs_r(const struct lys_module *ly_mod)
     return NULL;
 }
 
+static sr_error_info_t *
+sr_shm_prefix(const char **prefix)
+{
+    sr_error_info_t *err_info = NULL;
+
+    *prefix = getenv(SR_SHM_PREFIX_ENV);
+    if (*prefix == NULL) {
+        *prefix = SR_SHM_PREFIX_DEFAULT;
+    } else if (strchr(*prefix, '/') != NULL) {
+        *prefix = NULL;
+        sr_errinfo_new(&err_info, SR_ERR_INVAL_ARG, NULL,
+            SR_SHM_PREFIX_ENV" cannot contain slashes.");
+    }
+
+    return err_info;
+}
+
+sr_error_info_t *
+sr_path_main_shm(char **path)
+{
+    sr_error_info_t *err_info = NULL;
+    const char *prefix;
+
+    err_info = sr_shm_prefix(&prefix);
+    if (err_info) {
+        goto end;
+    }
+
+    if (asprintf(path, "/%s_main", prefix) == -1) {
+        SR_ERRINFO_MEM(&err_info);
+        *path = NULL;
+    }
+
+end:
+    return err_info;
+}
+
+sr_error_info_t *
+sr_path_ext_shm(char **path)
+{
+    sr_error_info_t *err_info = NULL;
+    const char *prefix;
+
+    err_info = sr_shm_prefix(&prefix);
+    if (err_info) {
+        goto end;
+    }
+
+    if (asprintf(path, "/%s_ext", prefix) == -1) {
+        SR_ERRINFO_MEM(&err_info);
+        *path = NULL;
+    }
+
+end:
+    return err_info;
+}
+
 sr_error_info_t *
 sr_path_sub_shm(const char *mod_name, const char *suffix1, int64_t suffix2, int abs_path, char **path)
 {
     sr_error_info_t *err_info = NULL;
+    const char *prefix;
     int ret;
 
+    err_info = sr_shm_prefix(&prefix);
+    if (err_info) {
+        return err_info;
+    }
+
     if (suffix2 > -1) {
-        ret = asprintf(path, "%s/srsub_%s.%s.%08x", abs_path ? SR_SHM_DIR : "", mod_name, suffix1, (uint32_t)suffix2);
+        ret = asprintf(path, "%s/%ssub_%s.%s.%08x",
```

Comment:
The convention I used is that in case a line is too long (more than 120 chars), I continue on the next line starting with double indentation (8 spaces). There are several similar places where this is different.

Reply:
done